### PR TITLE
add/fix function breakpoints

### PR DESCRIPTION
**Pull request by hoehrmann**
With this function breakpoints seem to work okay ~~, except for the linked bug in vscode~~ .

### CODE REVIEW

**Comment by raix** [comment]
@hoehrmann looks great! I looked at the test failing on mac perl 5.14 - not sure what the issue is, any ideas?

**Comment by hoehrmann** [comment]
@raix: In https://travis-ci.org/raix/vscode-perl-debug/builds/498897430 Perl 5.14 and Perl 5.16 on Linux failed, and the difference between the two is just in comments and the readme, no code changes if I am not mistaken, so those seem to be partly non-deterministic failures that I suspect have other causes than the changes here. Perl 5.14 is 8 years old, Perl 5.16 is 7 years old, if it were my call, I would probably remove them from the test matrix.

**Comment by raix** [comment]
Ok - let's comment them out for now

**Comment by hoehrmann** [comment]
@raix: I adapted them with the last commit and they pass with all versions now.

**Comment by raix** [comment]
Thanks @hoehrmann !
Sound like the extension interface / vs code is not ready - or some other race condition?

**Comment by raix** [comment]
@hoehrmann is it because vs code is actually sending breakpoint data before the debugger sends the initialized event? (I think we already do have a command queue in the adapter - just wondering what could be the cause of this)

**Comment by hoehrmann** [comment]
@raix: See the new issue, I came to the same conclusion trying to debug why my breakpoints were having no effect when set before starting a debug session in vscode.

File: src/tests/adapter.test.ts [hunk 1]
```diff
@@ -150,6 +150,69 @@ describe('Perl debug Adapter', () => {
 		});
 	});
 
+	describe.skip('setFunctionBreakpoints', () => {
```
Comment (raix) [inline src/tests/adapter.test.ts]: is this test skipped on purpose?

Comment (hoehrmann) [reply src/tests/adapter.test.ts]: Yes, due to issue #75.

**Comment by hoehrmann** [comment]
Retracted in favour of https://github.com/raix/vscode-perl-debug/pull/77

**Pull request closed without merging**